app/services/indexer.rb

```diff
@@ -0,0 +1,92 @@
+class Indexer
```

Comment:
this class specialises in indexing `datasets`, so might be worth renaming this class to something like `dataset_indexer`.